src/Xz/XzBinaryLocator.php

```diff
@@ -2,36 +2,14 @@
 
 namespace rikudou\SkQrPayment\Xz;
 
-use rikudou\SkQrPayment\Exception\QrPaymentException;
-
 final class XzBinaryLocator implements XzBinaryLocatorInterface
 {
-    /**
-     * @var string|null
-     */
-    private $path;
-
-    public function __construct(?string $path)
-    {
-        $this->path = $path;
-    }
-
     public function getXzBinary(): string
     {
-        if ($this->path === null) {
-            exec('which xz', $output, $return);
-            if ($return !== 0) {
-                throw new QrPaymentException("'xz' binary not found in PATH, specify it using setXzBinary()");
-            }
-            if (!isset($output[0])) {
-                throw new QrPaymentException("'xz' binary not found in PATH, specify it using setXzBinary()");
-            }
-            $this->path = $output[0];
-        }
-        if (!file_exists($this->path)) {
-            throw new QrPaymentException("The path '{$this->path}' to 'xz' binary is invalid");
-        }
+        $error = sprintf('The class "%s" is deprecated, please use "%s" or "%s"', XzBinaryLocator::class, LinuxXzBinaryLocator::class, WindowsXzBinaryLocator::class);
+        trigger_error($error, E_USER_DEPRECATED);
+        class_alias(LinuxXzBinaryLocator::class, XzBinaryLocator::class);
```

Comment:
This needs to be outside class, at the top of the file so it gets triggered the first time someone loads this class, I think this would not even work and throw an error because you are trying to redefine class, you're also returning the error instead of the path, the idea behing backwards compatibility is that you cannot break the code for your users - meaning that if someone specifically used the class `XzBinaryLocator` they should be able to continue using it without changing the functionality.

Delete the whole class and leave only the class_alias part. Since the class `XzBinaryLocator` won't actually exist anymore you will probably have to use the FQN 'rikudou\SkQrPayment\Xz\XzBinaryLocator' instead of `XzBinaryLocator::class`. Also please don't use the `$error` variable, it would create a global variable since it will be out of any class and I'd like to avoid it, use the `sprintf()` call directly as an argument.